hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/entity/AccountBuilderImpl.java

```diff
@@ -0,0 +1,284 @@
+
+/**
+ * An implementation of {@link AccountBuilder} for building Account instances. This class is
+ * <strong>not</strong> exported from the module. FUTURE: Should be moved to token-service-impl
+ * module
+ */
+public class AccountBuilderImpl implements AccountBuilder {
+    private Account copyOf;
+
+    // These fields are the ones that can be set in the builder
+    // FUTURE: Replace the empty KeyList we use for 0.0.800 and hollow
+    // accounts with null
+    private HederaKey key;
+    private long expiry;
+    private long balance;
+    private String memo;
+    private boolean deleted;
+    private boolean receiverSigRequired;
+    private long numberOfOwnedNfts;
+    private int maxAutoAssociations;
+    private int usedAutoAssociations;
+    private int numAssociations;
+    private int numPositiveBalances;
+    private long ethereumNonce;
+    private long stakedToMe;
+    private long stakePeriodStart;
+    private long stakedNum;
+    private boolean declineReward;
+    private long stakeAtStartOfLastRewardedPeriod;
+    private long autoRenewAccountNumber;
+    private long autoRenewSecs;
+    private long accountNumber;
+    private byte[] alias = new byte[0];
+    private boolean isSmartContract;
+
+    /**
+     * Create a builder for creating {@link Account}s, using the given copy as the basis for all
+     * settings that are not overridden.
+     *
+     * @param copyOf The instance to copy
+     */
+    public AccountBuilderImpl(@NonNull Account copyOf) {
+        this.copyOf = Objects.requireNonNull(copyOf);
+        this.key = copyOf.getKey().isEmpty() ? null : copyOf.getKey().get();
+        this.expiry = copyOf.expiry();
+        this.balance = copyOf.balanceInTinyBar();
+        this.memo = copyOf.memo();
+        this.deleted = copyOf.isDeleted();
+        this.receiverSigRequired = copyOf.isReceiverSigRequired();
+        this.numberOfOwnedNfts = copyOf.numberOfOwnedNfts();
+        this.maxAutoAssociations = copyOf.maxAutoAssociations();
+        this.usedAutoAssociations = copyOf.usedAutoAssociations();
+        this.numAssociations = copyOf.numAssociations();
+        this.numPositiveBalances = copyOf.numPositiveBalances();
+        this.ethereumNonce = copyOf.ethereumNonce();
+        this.stakedToMe = copyOf.stakedToMe();
+        this.stakePeriodStart = copyOf.stakePeriodStart();
+        this.stakedNum = copyOf.stakedNum();
+        this.declineReward = copyOf.declineReward();
+        this.stakeAtStartOfLastRewardedPeriod = copyOf.stakeAtStartOfLastRewardedPeriod();
+        this.autoRenewAccountNumber = copyOf.autoRenewAccountNumber();
+        this.autoRenewSecs = copyOf.autoRenewSecs();
+        this.accountNumber = copyOf.accountNumber();
+        this.alias = copyOf.getAlias().isEmpty() ? new byte[0] : copyOf.getAlias().get();
+        this.isSmartContract = copyOf.isSmartContract();
+    }
+
+    public AccountBuilderImpl() {
+        /* Default constructor for creating new Accounts */
```

Comment:
Does it make sense to define other default values for some of the fields?

Reply:
Done !